src/dash-adapter.js

```diff
@@ -307,6 +346,95 @@ export default class DashAdapter extends BaseMediaSourceAdapter {
     }
   }
 
+  /**
+   * apply ABR restrictions
+   * @private
+   * @returns {void}
+   */
+  _maybeApplyAbrRestrictions(): void {
+    if (this._config.capLevelToPlayerSize) {
+      const videoTracks = this._getVideoTracks();
+      const getMinDimensions = (dim): number => Math.min.apply(null, videoTracks.map(variant => variant[dim]));
+      //Get minimal allowed dimensions
+      const minWidth = getMinDimensions('width');
+      const minHeight = getMinDimensions('height');
+      const updateAbrRestrictions = () => {
```

Comment:
it's a long function, maybe put it outside `_maybeApplyAbrRestrictions` ?

Reply:
this would probably find itself outside in abrManager eventually. not sure we need premature optimisations